docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.15.3.rst

```diff
@@ -0,0 +1,199 @@
+
+Python
+""""""
+
+*   Added modeling of more :code:`FileSystemAccess` in packages :code:`cherrypy`, :code:`aiofile`, :code:`aiofiles`, :code:`anyio`, :code:`sanic`, :code:`starlette`, :code:`baize`, and :code:`io`. This will mainly affect the *Uncontrolled data used in path expression* (:code:`py/path-injection`) query.
+
+Swift
+"""""
+
+*   Added additional sinks for the "Uncontrolled data used in path expression" (:code:`swift/path-injection`) query. Some of these sinks are heuristic (imprecise) in nature.
+*   Fixed an issue where some Realm database sinks were not being recognized for the :code:`swift/cleartext-storage-database` query.
+
+New Queries
+~~~~~~~~~~~
+
+Swift
+"""""
+
+*   Added new query "System command built from user-controlled sources" (:code:`swift/command-line-injection`) for Swift. This query detects system commands built from user-controlled sources without sufficient validation. The query was previously `contributed to the 'experimental' directory by @maikypedia <https://github.com/github/codeql/pull/13726>`__ but will now run by default for all code scanning users.
+*   Added a nw query "Missing regular expression anchor" (:code:`swift/missing-regexp-anchor`) for Swift. This query detects regular expressions without anchors that can be vulnerable to bypassing.
+
+Query Metadata Changes
+~~~~~~~~~~~~~~~~~~~~~~
+
+JavaScript/TypeScript
+"""""""""""""""""""""
+
+*   Lower the severity of log-injection to medium.
+*   Increase the severity of XSS to high.
+
+Language Libraries
+------------------
+
+Bug Fixes
+~~~~~~~~~
+
+Golang
+""""""
+
+*   A bug has been fixed that meant that value flow through an array was not tracked correctly in some circumstances. Taint flow was tracked correctly.
+
+Breaking Changes
+~~~~~~~~~~~~~~~~
+
+C/C++
+"""""
+
+*   The expressions :code:`AssignPointerAddExpr` and :code:`AssignPointerSubExpr` are no longer subtypes of :code:`AssignBitwiseOperation`.
+
+Major Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Swift
+"""""
+
+*   Added Swift 5.9.1 support
+*   New AST node is extracted: :code:`SingleValueStmtExpr`
+
+Minor Analysis Improvements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+C/C++
+"""""
+
+*   The "Returning stack-allocated memory" (:code:`cpp/return-stack-allocated-memory`) query now also detects returning stack-allocated memory allocated by calls to :code:`alloca`, :code:`strdupa`, and :code:`strndupa`.
+*   Added models for :code:`strlcpy` and :code:`strlcat`.
+*   Added models for the :code:`sprintf` variants from the :code:`StrSafe.h` header.
+*   Added SQL API models for :code:`ODBC`.
+*   Added taint models for :code:`realloc` and related functions.
+
+C#
+""
+
+*   The predicate :code:`UnboundGeneric::getName` now prints the number of type parameters as a :code:` `N` suffix, instead of a :code:`<,...,>` suffix. For example, the unbound generic type
```

Comment:
It looks as if something's gone wrong here. Possibly it should be?

```suggestion
*   The predicate :code:`UnboundGeneric::getName` now prints the number of type parameters as a :code:`N` suffix, instead of a :code:`<,...,>` suffix. For example, the unbound generic type
```

Reply:
The bactick was actually meant to be included. So an `` `N`` suffix (don't know how to get rid of the leading space).

Reply:
@hvitved Thanks. I fixed a bug in the generator so that it will now produce correct RST even if there's a backtick and even if there's a leading or trailing space (the latter will be removed automatically). So it looks correct on the CodeQL website, even if it's a bit odd in markdown: https://codeql.github.com/docs/codeql-overview/codeql-changelog/codeql-cli-2.15.3/#id7

Reply:
We've corrected it in the [published copy on the CodeQL site](https://codeql.github.com/docs/codeql-overview/codeql-changelog/codeql-cli-2.15.3/#id7) manually. Pierre is going to look into the process that converts it from markdown into RST because, after further investigation, it looks as if that's where the problem is.

Reply:
Since it's fixed in the copy and in the tool now, I'll resolve this